### ISSUE TITLE
Add per scrape sample_size_limit configuration option to vmagent

### ISSUE THREAD

**Issue by stevenbeaullieu** [VictoriaMetrics/VictoriaMetrics#6429]
### Is your feature request related to a problem? Please describe

We are having to set a fairly large `-promscrape.maxScrapeSize` to accomodate a large response size from the vmalert metrics endpoint due to us running a very large number of recording rules and alerts.  It would be nice to have the global response size limit be smaller and only set a very large response size limit where needed.

### Describe the solution you'd like

Add a `sample_size_limit` configuration option to scrapes for vmagent that will override `-promscrape.maxScrapeSize` and fail the scrape and not ingest anything if the response exceeds the size.  That way you don't have to set a very large global max scrape size just for a few scrapes.

### Describe alternatives you've considered

Setting a very high `-promscrape.maxScrapeSize`, but this isn't ideal because we'd like to limit some customer scrapes to smaller sizes than we need for some scrapes related to our platform.

**Comment by AndrewChubatiuk** [comment]
Hey @stevenbeaullieu 
Added PR for this feature, need to wait for a feedback from a team
You can test it in a meanwhile

**Comment by stevenbeaullieu** [comment]
Thank you.  Had one more suggestion that sort of relates, but can post a new feature request if needed.  For the [Automatically generated metrics](https://docs.victoriametrics.com/vmagent/#automatically-generated-metrics) for vmagent scrapes, can there be a new metric that is something like "scrape_response_size_bytes" that returns the response size of the scraped metrics in bytes, to help do analysis for which jobs might be close to the global limit and potentially need a per scrape override?

**Comment by AndrewChubatiuk** [comment]
> Thank you. Had one more suggestion that sort of relates, but can post a new feature request if needed. For the [Automatically generated metrics](https://docs.victoriametrics.com/vmagent/#automatically-generated-metrics) for vmagent scrapes, can there be a new metric that is something like "scrape_response_size_bytes" that returns the response size of the scraped metrics in bytes, to help do analysis for which jobs might be close to the global limit and potentially need a per scrape override?

added to the same PR

**Comment by hagen1778** [comment]
https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6434 has been merged and will be likely available in the next release.

**Comment by hagen1778** [comment]
@stevenbeaullieu ability to limit max scrape size is available starting from [v1.102.0-rc2](https://docs.victoriametrics.com/changelog/#v11020-rc2)
See [this doc](https://docs.victoriametrics.com/sd_configs/#scrape_configs) for details:
```
  # max_scrape_size is an optional parameter for limiting the response size in bytes from scraped targets.
  # By default, uses limit from -promscrape.maxScrapeSize command-line flag.
  # Example values:
  # - "10MiB" - 10 * 1024 * 1024 bytes
  # - "100MB" - 100 * 1000 * 1000 bytes
  #
  # max_scrape_size: <size>
```
vmagent now also [generates](https://docs.victoriametrics.com/vmagent/#automatically-generated-metrics) `scrape_response_size_bytes` for each scraped target.
